process/block/miniblocks/miniblocks_processor.go

```diff
@@ -0,0 +1,74 @@
+package miniblocks
+
+import (
+	"github.com/ElrondNetwork/covalent-indexer-go"
+	"github.com/ElrondNetwork/covalent-indexer-go/schema"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go-core/hashing"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+var log = logger.GetOrCreate("process/block/miniBlocks/miniBlocksProcessor")
+
+type miniBlocksProcessor struct {
+	hasher     hashing.Hasher
+	marshaller marshal.Marshalizer
+}
+
+// NewMiniBlocksProcessor will create a new instance of miniBlocksProcessor
+func NewMiniBlocksProcessor(hasher hashing.Hasher, marshalizer marshal.Marshalizer) (*miniBlocksProcessor, error) {
+	if check.IfNil(marshalizer) {
+		return nil, covalent.ErrNilMarshaller
+	}
+	if check.IfNil(hasher) {
+		return nil, covalent.ErrNilHasher
+	}
+
+	return &miniBlocksProcessor{
+		hasher:     hasher,
+		marshaller: marshalizer,
+	}, nil
+}
+
+// ProcessMiniBlocks converts mini blocks core data to a specific mini blocks structure array defined by avro schema
+func (mbp *miniBlocksProcessor) ProcessMiniBlocks(header data.HeaderHandler, body data.BodyHandler) ([]*schema.MiniBlock, error) {
+	erdBody, castOk := body.(*block.Body)
+	if !castOk {
+		return nil, covalent.ErrBlockBodyAssertion
+	}
+
+	miniBlocks := make([]*schema.MiniBlock, 0)
```

Comment:
`miniBlocks := make([]*schema.MiniBlock, 0, len(erdMiniBlocks))` -- in order to allocate memory for the new slice

Reply:
Done